util/topgen/intermodule.py

```diff
@@ -0,0 +1,111 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+from typing import Dict
+
+from .lib import *
+
+
+def elab_intermodule(topcfg):
+    """Check the connection of inter-module and categorize them
+
+    In the top template, it uses updated inter_module fields to create
+    connections between the modules (incl. memories). This function is to
+    create and check the validity of the connections `inter_module` using IPs'
+    `inter_signal_list`.
+    """
+
+    list_of_intersignals = []
+
+    if "inter_signal" not in topcfg:
+        topcfg["inter_signal"] = {}
```

Comment:
is there a case where we'd have `topcfg[inter_signal]`?  I thought top would only have inter_module, and all the inter signals would be defined in either the IPs or the memories.  Or is this maybe a first step towards absorbing alerts and interrupts?

Reply:
`inter_signal` field is generated by topgen. It has all the inter_signal lists and also it's top-level definitions. So that the tool can easily find what is missing.

Reply:
ah okay that makes sense.